apps/browser/settings/searchenginemodel.cpp

```diff
@@ -68,3 +68,46 @@ void SearchEngineModel::componentComplete()
 {
 
 }
+
+QStringList SearchEngineModel::searchEngines()
+{
+    QStringList engines;
+    for (int i = 0; i < m_searchEngines.count(); i++)
+        engines.append(m_searchEngines[i].title);
+    return engines;
+}
+
+void SearchEngineModel::addSearchEngine(const QString& title, const QString& url)
+{
+    bool found = false;
+    for (int i = 0; i < m_searchEngines.count(); i++) {
+        if (m_searchEngines[i].title == title) {
+            found = true;
+            break;
+        }
+    }
+    if (!found) {
+        SearchEngine engine(url, title, false);
+        m_searchEngines.append(engine);
+        emit countChanged();
+        emit searchEnginesChanged();
```

Comment:
As the searchEngines string list is only for "contains" => this should not be needed either.